dist/src/main/java/io/camunda/zeebe/shared/IdleStrategyConfig.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.shared;
+
+import io.camunda.zeebe.scheduler.ActorScheduler.ActorSchedulerBuilder;
+import io.camunda.zeebe.shared.IdleStrategyConfig.IdleStrategyProperties;
+import java.time.Duration;
+import java.util.function.Supplier;
+import org.agrona.concurrent.BackoffIdleStrategy;
+import org.agrona.concurrent.IdleStrategy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.Nullable;
+
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties(IdleStrategyProperties.class)
+public final class IdleStrategyConfig {
+  private final IdleStrategyProperties properties;
```

Comment:
❓ Why we need all of these wrapping code, sorry I got a bit lost

Reply:
Could we just create the backoff idle strategy with the properties without creating another supplier?

Reply:
It needs to be a supplier because otherwise you would share the same idle strategy for every thread :(

Reply:
AH ok makes sense thanks. Could we add somewhere a comment why this is like it